saleor/tests/e2e/shop/utils/preparing_shop.py

```diff
@@ -1,44 +1,185 @@
+import base64
+
 from ...channel.utils import create_channel
-from ...shipping_zone.utils.shipping_method import create_shipping_method
-from ...shipping_zone.utils.shipping_method_channel_listing import (
+from ...shipping_zone.utils import (
+    create_shipping_method,
     create_shipping_method_channel_listing,
+    create_shipping_zone,
+)
+from ...taxes.utils import (
+    create_tax_class,
+    get_tax_configurations,
+    update_tax_class,
+    update_tax_configuration,
 )
-from ...shipping_zone.utils.shipping_zone import create_shipping_zone
 from ...warehouse.utils import create_warehouse
+from ..utils.shop_update_settings import update_shop_settings
+
+
+def decode_and_modify_base64_descriptor(encoded_string, new_descriptor=None):
+    base64_bytes = encoded_string.encode("ascii")
+    decoded_bytes = base64.b64decode(base64_bytes)
+    decoded_string = decoded_bytes.decode("ascii")
+
+    modified_string = decoded_string
+    if new_descriptor:
+        parts = decoded_string.split(":")
+        if len(parts) >= 2:
+            modified_string = f"{new_descriptor}:{parts[1]}"
+
+    modified_base64 = base64.b64encode(modified_string.encode("ascii")).decode("ascii")
+
+    padding = "=" * ((4 - len(modified_base64) % 4) % 4)
+    modified_base64_padded = modified_base64 + padding
+
+    return decoded_string, modified_base64_padded
+
+
+def create_and_update_tax_classes(e2e_staff_api_client, tax_rates):
+    created_tax_classes = {rate["type"]: {} for rate in tax_rates}
+
+    for tax_rate_info in tax_rates:
+        tax_class_data = create_tax_class(
+            e2e_staff_api_client,
+            tax_rate_info["name"],
+            [
+                {
+                    "countryCode": tax_rate_info["country_code"],
+                    "rate": tax_rate_info["rate"],
+                }
+            ],
+        )
+        created_tax_classes[tax_rate_info["type"]] = {
+            "id": tax_class_data["id"],
+            "rate": tax_rate_info["rate"],
+        }
+
+        tax_class_update_input = {
+            "updateCountryRates": [
+                {
+                    "countryCode": tax_rate_info["country_code"],
+                    "rate": tax_rate_info["rate"],
+                }
+            ]
+        }
+        update_tax_class(
+            e2e_staff_api_client, tax_class_data["id"], tax_class_update_input
+        )
+
+    return created_tax_classes
+
+
+def prepare_shop(
+    e2e_staff_api_client,
+    warehouses=None,
+    channels=None,
```

Comment:
I'll also suggest renaming `channels` to some other name (f.e. `channels_settings`), as the `channels` suggests that you are providing channel instances in this parameter